test/functional/test_framework/util.py

```diff
@@ -56,6 +56,10 @@ def assert_equal(thing1, thing2, *args):
     if thing1 != thing2 or any(thing1 != arg for arg in args):
         raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
 
+def assert_not_equal(thing1, thing2, *args):
+    if thing1 == thing2 or any(thing1 == arg for arg in args):
+        raise AssertionError("%s" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
```

Comment:
This error message doesn't make sense when `*args` is non-empty: if you say `assert_not_equal(a, 1, 2)` and `a` is 1 or 2, it will report `1 == 1 == 2` or `2 == 1 == 2`.

Reply:
I've updated this to instead only take three arguments and the last one being an optional message since I did not see anywhere we were already asserting `!=` in a chain.

This should clear up the message and allow the one location we do use a message to be used

updated in [4488120](https://github.com/bitcoin/bitcoin/pull/29500/commits/4488120ab7db775ce0ae8a73f910dddca1dee123)